peaq/utils.py

```diff
@@ -0,0 +1,242 @@
+            self.substrate, module, extrinsic, params))
+
+    # TODO
+    def execute(self, wait_for_finalization=False, alt_keypair=None) -> str:
+        """Executes the extrinsic-stack"""
+        if not self.batch:
+            return ''
+        if alt_keypair is None:
+            alt_keypair = self.keypair
+        return _execute_extrinsic_batch(
+            self.substrate, alt_keypair, self.batch, wait_for_finalization)
+
+    # TODO
+    def execute_n_clear(self, alt_keypair=None, wait_for_finalization=False) -> str:
+        """Combination of execute() and clear()"""
+        if alt_keypair is None:
+            alt_keypair = self.keypair
+        receipt = self.execute(wait_for_finalization, alt_keypair)
+        self.clear()
+        return receipt
+
+    def clear(self):
+        """Clears the current extrinsic-stack"""
+        self.batch = []
+
+    def clone(self, keypair_or_uri=None):
+        """Creates a duplicate, by using the same SubstrateInterface"""
+        if keypair_or_uri is None:
+            keypair_or_uri = self.keypair
+        return ExtrinsicBatch(self.substrate, keypair_or_uri)
+
+
+def _compose_call(substrate, module, extrinsic, params):
+    """
+    Composes a substrate-extrinsic-call on any module
+    Example:
+      module = 'Rbac'
+      extrinsic = 'add_role'
+      params = {'role_id': entity_id, 'name': name }
+    """
+    return substrate.compose_call(
+        call_module=module,
+        call_function=extrinsic,
+        call_params=params
+    )
+
+
+def _compose_sudo_call(substrate, module, extrinsic, params):
+    """
+    Composes a substrate-sudo-extrinsic-call on any module
+    Parameters same as in compose_call, see above
+    """
+    payload = _compose_call(substrate, module, extrinsic, params)
+    return _compose_call(substrate, 'Sudo', 'sudo', {'call': payload.value})
+
+
+def _execute_extrinsic_batch(substrate, kp_src, batch,
+                             wait_for_finalization=False) -> str:
+    """
+    Executes a extrinsic-stack/batch-call on substrate
+    Parameters:
+      substrate:  SubstrateInterface
+      kp_src:     Keypair
+      batch:      list[_compose_call(), _compose_call(), ...]
+    """
+    # Wrap payload into a utility batch cal
+    call = substrate.compose_call(
+        call_module='Utility',
+        call_function='batch_all',
+        call_params={
+            'calls': batch,
+        })
+
+    nonce = substrate.get_account_nonce(kp_src.ss58_address)
+    extrinsic = substrate.create_signed_extrinsic(
+        call=call,
+        keypair=kp_src,
+        era={'period': 64},
+        nonce=nonce
+    )
+
+    receipt = substrate.submit_extrinsic(
+        extrinsic, wait_for_inclusion=True,
+        wait_for_finalization=wait_for_finalization)
+    if len(batch) == 1:
+        description = _generate_call_description(batch[0])
+    else:
+        description = _generate_batch_description(batch)
+    if DEBUG:
```

Comment:
I would prefer, to show this any time, not only when DEBUG is set - because it gives a better response/feedback when executing all integration tests and while waiting for them to finish...

Reply:
If we talk about the SDK, we shouldn't print any information because users may use it elsewhere. Imagine that you are using a 3rd party library, and it always shows the information. And the worser scenario is users may want to capture the stdout when doing things.

Therefore, I prefer not to show anything. TBH, it's better to use the environment argument to handle that.